input/postgres/statements.go

```diff
@@ -234,12 +237,23 @@ func GetStatements(server *state.Server, logger *util.Logger, db *sql.DB, global
 		if showtext {
 			statementTexts[key] = receivedQuery.String
 		}
+		if queryID.Valid && showtext {
+			if identity, ok := server.QueryIdentities[queryID.Int64]; ok {
+				identity.LastSeen = now
+			} else {
+				server.QueryIdentities[queryID.Int64] = state.QueryIdentity{
+					Fingerprint: util.FingerprintQuery(receivedQuery.String, server.Config.FilterQueryText, -1),
+					LastSeen:    now,
+				}
+			}
+		}
 		if ignoreIOTiming(postgresVersion, receivedQuery) {
 			stats.BlkReadTime = 0
 			stats.BlkWriteTime = 0
 		}
 		statementStats[key] = stats
 	}
+	server.QueryIdentitiesMutex.Unlock()
```

Comment:
Thanks for moving these around. I think that looks good, however we also need to make sure to unlock the mutex in error cases where we do an early return out of the loop (i.e. line 222).

Reply:
Ah, good catch